astropy/modeling/core.py

```diff
@@ -678,6 +678,57 @@ def __init__(self, *args, meta=None, name=None, **kwargs):
         # Parameter values must be passed in as keyword arguments in order to
         # distinguish them
         self._initialize_parameters(args, kwargs)
+        self._initialize_unit_support()
+
+    def _initialize_unit_support(self):
+        """
+        Convert self._input_units_strict and
+        self.input_units_allow_dimensionless to dictionaries
+        mapping input name to a boolena value.
+        """
+        if isinstance(self._input_units_strict, bool):
+            self._input_units_strict = {key: self._input_units_strict for
+                                        key in self.__class__.inputs}
+
+        if isinstance(self._input_units_allow_dimensionless, bool):
+            self._input_units_allow_dimensionless = {key: self._input_units_allow_dimensionless
+                                                     for key in self.__class__.inputs}
+
+    @property
+    def input_units_strict(self):
+        """
+        Enforce strict units on inputs to evaluate. If this is set to True, input
+        values to evaluate have to be in the exact right units specified by
+        input_units. In this case, if the input quantities are convertible to
+        input_units, they are converted. If this is a dictionary then it should
+        map input name to a bool to set strict input units for that parameter.
+        """
+        return self._input_units_strict
+
+    @input_units_strict.setter
+    def input_units_strict(self, val):
+        if isinstance(val, bool):
+            self._input_units_strict = {key: val for key in self.__class__.inputs}
+        else:
+            self._input_units_strict = val
+
+    @property
+    def input_units_allow_dimensionless(self):
+        """
+        Allow dimensionless input (and corresponding output). If this is True,
+        input values to evaluate will gain the units specified in input_units. If
+        this is a dictionary then it should map input name to a bool to allow
+        dimensionless numbers for that input.
+        Only has an effect if input_units is defined.
+        """
+        return self._input_units_allow_dimensionless
+
+    @input_units_allow_dimensionless.setter
+    def input_units_allow_dimensionless(self, val):
+        if isinstance(val, bool):
+            self._input_units_allow_dimensionless = {key: val for key in self.__class__.inputs}
+        else:
+            self._input_units_allow_dimensionless = val
 
```

Comment:
The exmple below fails. Am I misunderstanding that it should work:
```
g = Gaussian1D(10*u.Hz, 3*u.m, 2*u.m)
g.input_units_allow_dimensionless = True
g(1)

```

Reply:
I would think this should pass yes.

Reply:
Actually no, the reason this is failing is because the parameters have units (and `x - mean` is in the evaulate). `input_units_allow_dimensionless` is set here and it allows the dimensionless input, but it does not strip the units from the paramters.

Reply:
Yes, it is the reason for the failure. However, the docstrings states that in this case the input gains the units of the model so I expect the input to become 1*u.m and not fail. How should `input_units_allow_unitless=True` behave in this case?